crates/swc_ecma_lints/src/rules/no_alert.rs

```diff
@@ -0,0 +1,153 @@
+use crate::{
+    config::{LintRuleReaction, RuleConfig},
+    rule::{visitor_rule, Rule},
+};
+use swc_common::{collections::AHashSet, errors::HANDLER, Span, SyntaxContext};
+use swc_ecma_ast::*;
+use swc_ecma_utils::{collect_decls_with_ctxt, ident::IdentLike};
+use swc_ecma_visit::{noop_visit_type, Visit, VisitWith};
+
+const FN_NAMES: &'static [&str] = &["alert", "confirm", "prompt"];
+const GLOBAL_THIS_PROP: &'static str = "globalThis";
+const OBJ_NAMES: &'static [&str] = &["window", GLOBAL_THIS_PROP];
+
+pub fn no_alert(
+    program: &Program,
+    config: &RuleConfig<()>,
+    top_level_ctxt: SyntaxContext,
+    es_version: EsVersion,
+) -> Option<Box<dyn Rule>> {
+    let top_level_declared_vars: AHashSet<Id> = collect_decls_with_ctxt(program, top_level_ctxt);
+    let rule_reaction = config.get_rule_reaction();
+
+    match rule_reaction {
+        LintRuleReaction::Off => None,
+        _ => Some(visitor_rule(NoAlert::new(
+            *rule_reaction,
+            top_level_declared_vars,
+            top_level_ctxt,
+            es_version,
+        ))),
+    }
+}
+
+#[derive(Debug, Default)]
+struct NoAlert {
+    expected_reaction: LintRuleReaction,
+    top_level_ctxt: SyntaxContext,
+    top_level_declared_vars: AHashSet<Id>,
+    pass_call_on_global_this: bool,
+    fn_names: AHashSet<&'static str>,
+    obj_names: AHashSet<&'static str>,
+    call_expr_span: Option<Span>,
+}
+
+impl NoAlert {
+    fn new(
+        expected_reaction: LintRuleReaction,
+        top_level_declared_vars: AHashSet<Id>,
+        top_level_ctxt: SyntaxContext,
+        es_version: EsVersion,
+    ) -> Self {
+        Self {
+            expected_reaction,
+            top_level_ctxt,
+            top_level_declared_vars,
+            pass_call_on_global_this: es_version < EsVersion::Es2020,
+            fn_names: FN_NAMES.iter().map(|n| *n).collect(),
+            obj_names: OBJ_NAMES.iter().map(|n| *n).collect(),
```

Comment:
Same here, we can just use `static` item from rule.

Reply:
You're mean that I can use static hashset?

Reply:
No, you don't need hastset for this.